equinox/_module/_flatten.py

```diff
@@ -0,0 +1,234 @@
+"""Utilities for generating optimized flatten/unflatten functions for Module."""
+
+import dataclasses
+import textwrap
+from enum import Enum
+from typing import Any, Final, Literal, TypeVar
+
+import jax.tree_util as jtu
+
+
+# Names for generated functions
+FLAT_FUNC_NAME = "<generated_flatten_{0}>"
+FLAT_KEYS_NAME = "<generated_flatten_with_keys_{0}>"
+UNFLAT_NAME = "<generated_unflatten_{0}>"
+
+# Fields of Module wrappers to always include
+_WRAPPER_FIELD_NAMES: Final = (
+    "__module__",
+    "__name__",
+    "__qualname__",
+    "__doc__",
+    "__annotations__",
+)
+
+# Indentation for generated code
+INDENT: Final = " " * 4
+
+
+# Sentinel values for flattening/unflattening
+class Sentinel(Enum):
+    """Sentinel values for flattening/unflattening."""
+
+    MISSING = "MISSING"
+
+
+MISSING = Sentinel.MISSING
+
+# Code template for flattening the wrapper fields
+FLAT_WRAPPERS = ", ".join(
```

Comment:
Maybe `GET_WRAPPERS` (or `_GET_WRAPPERS`) for parity with the `SET_` flavour below?